CMakeLists.txt

```diff
@@ -60,18 +60,15 @@ set(VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}${VERSION_REV})
 set(PACKAGE ${CMAKE_PROJECT_NAME})
 string(TOUPPER "${CMAKE_PROJECT_NAME}" UPCASE_PROJECT_NAME)
 
-# Top level cmake file, set options
-if(NOT NTTEC_USE_SIMD)
-  set(NTTEC_USE_SIMD "OFF"
-    CACHE
-    STRING
-    "Build with SIMD vectorizations."
-    FORCE
-  )
-  set_property(CACHE NTTEC_USE_SIMD PROPERTY STRINGS
-    "OFF" "SSE4" "AVX2"
-  )
-endif()
+# Setting for SIMD level
```

Comment:
About the commit message (since I can't comment directly on it…)

> If NTTEC_USE_SIMD is not given or given but not supported, default value
is used.

IMHO, if USE_SIMD is not defined, then SIMD should be disabled.
You should auto-detect the best SIMD instruction set only if USE_SIMD is enabled I think.